FreeRTOS-Plus/Test/CBMC/proofs/prvProcessEthernetPacket/prvProcessEthernetPacket_harness.c

```diff
@@ -0,0 +1,84 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_UDP_IP.h"
+#include "FreeRTOS_DHCP.h"
+#include "NetworkInterface.h"
+#include "NetworkBufferManagement.h"
+#include "FreeRTOS_ARP.h"
+
+#include "cbmc.h"
+
+
+/****************************************************************
+ * Signature of function under test
+ ****************************************************************/
+
+void prvProcessEthernetPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer );
+
+
+/* This proof was done before. Hence we assume it to be correct here. */
+void vARPRefreshCacheEntry( const MACAddress_t * pxMACAddress, const uint32_t ulIPAddress ) { }
+
+
+/* Implementation of safe malloc */
+void *safeMalloc(size_t xWantedSize ){
+	if(xWantedSize == 0){
+		return NULL;
+	}
+	uint8_t byte = nondet_uint8_t();
+	return byte ? malloc(xWantedSize) : NULL;
```

Comment:
Could we just make the whole function "return nondet_byte() ? malloc(xWantedSize) : NULL"?

That "NULL on size 0" was an old, mistaken interpretation of the malloc spec, I think.  malloc is allowed to return a pointer on malloc(0) that can be freed as long as it is never dereferenced.

Reply:
I will do that.
But as far as malloc(0) is concerned, I think we should stick with returning NULL because that is what FreeRTOS does. See [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/master/portable/MemMang/heap_4.c#L115-L264).